UPGRADING.md

```diff
@@ -18,7 +18,24 @@ is typically found in `RegisterAPIRoutes`.
 
 ### AppModule Interface
 
-Remove `Querier`, `Route` and `LegacyQuerier` from the app module interface. This removes and fully deprecates all legacy queriers. All modules no longer support the REST API previously known as the LCD, and the `sdk.Msg#Route` method won't be used anymore.
+The `AppModule` interface is being refactored into `cosmossdk.io/core` and is now fully composed of extension interfaces
+based off of the `cosmossdk.io/core/appmodule.AppModule` tag interface. All `AppModule` implementations must now implement
+this tag interface which consists of the following code:
+
+```go
+// IsOnePerModuleType implements the depinject.OnePerModuleType interface.
+func (AppModule) IsOnePerModuleType() {}
+
+// IsAppModule implements the appmodule.AppModule interface.
+func (AppModule) IsAppModule() {}
```

Comment:
The plan is to put this in v0.47, right? I'm torn between the fact that this AppModule is my preferred approach, but at the same time it seems like a scope creep and ADR-061 hasn't been merged/approved yet by most people.